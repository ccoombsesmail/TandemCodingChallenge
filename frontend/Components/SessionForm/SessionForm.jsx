import React from 'react'
import styles from './SessionForm.module.css'
import { createUser, createSession } from '../../util/session_api_util'

class SessionForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: '',
      errors: null,
    }
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  update(type) {
    return (e) => {
      this.setState({ [type]: e.currentTarget.value })
    }
  }

  handleSubmit(e, formType) {
    const { username, password } = this.state
    e.preventDefault()
    if (formType === 'Sign Up') {
      createUser({ username, password })
        .then((res) => this.props.login(res.user))
        .fail((res) => {
          this.setState({ errors: res.responseJSON })
        })
    } else {
      createSession({ username, password })
        .then((res) => this.props.login(res.user))
        .fail((res) => {
          this.setState({ errors: res.responseJSON })
        })
    }
  }

  render() {
    const { toggleForm, formType } = this.props
    const { username, password, errors } = this.state
    return (
      <div className={styles.modalBg} onClick={toggleForm}>
        <div className={styles.modalChild} onClick={e => e.stopPropagation()}>
          <form className={styles.form} onSubmit={(e) => this.handleSubmit(e, formType)}>
            <h1>
              {formType}
            </h1>
            <label className={styles.formLabel}>
              <h4>Username</h4>
              <input maxLength={16} id="username" className={styles.formInput} type="text" value={username} onChange={this.update('username')} autoComplete="off" />
            </label>
            {
              errors && errors.usernameError ? <h3>{errors.usernameError}</h3> : null
            }
            <label className={styles.formLabel}>
              <h4>Password</h4>
              <input className={styles.formInput} type="password" value={password} onChange={this.update('password')} />
            </label>
            {
              errors && errors.passwordError ? <h3>{errors.passwordError}</h3> : null
            }
            <button type="submit">{formType}</button>
          </form>
        </div>
      </div>
    )
  }
}

export default SessionForm
