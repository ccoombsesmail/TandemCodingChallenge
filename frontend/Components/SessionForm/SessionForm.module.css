.modalBg {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.7);
  z-index: 100000;
}

.modalChild {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 10000;
  background-color: black;
  width: 500px;
  height: 400px;
  border-radius: 7px;
  -webkit-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
  -moz-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
  box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);

}

.form > h1 {
  margin-top: 50px;
  margin-bottom: 50px;
  color: white;
  letter-spacing: .5rem;
    font-size: 40px;


}

.form > h3 {
  margin: 5px;
  font-size: 18px;
  color: red;
}

.form {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #B13D6F;
  background-color: #973771;
  border-radius: 7px;


}

.form > button {
  background: #11998e;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #38ef7d, #11998e);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #38ef7d, #11998e); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  border: none;
  border-radius: 7px;
  padding: 7px;
  margin-top: 7px;
  width: 30%;
  font-weight: bolder;
  font-size: 17px;
  cursor: pointer;
  color: rgb(61, 4, 42);


}

.formLabel > h4 {
  margin-bottom: 5px;
  color: white;
  font-size: 20px;
}

.formLabel > input {
  width: 100%;
  height: 20px;
  border-radius: 10px;
  padding-left: 5px;

}

.formLabel {
  width: 60%;
  border-radius: 10px;
  margin-bottom: 5px;
}