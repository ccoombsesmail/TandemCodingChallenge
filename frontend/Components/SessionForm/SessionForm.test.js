import React from 'react'
import { shallow } from 'enzyme'
import SessionForm from './SessionForm'

describe('SessionForm', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<SessionForm debug />);

    expect(component).toMatchSnapshot();
  })
})