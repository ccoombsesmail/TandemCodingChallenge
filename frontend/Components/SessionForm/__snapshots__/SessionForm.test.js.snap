// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionForm should render correctly in "debug" mode 1`] = `
<div
  className="modalBg"
>
  <div
    className="modalChild"
    onClick={[Function]}
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <h1 />
      <label
        className="formLabel"
      >
        <h4>
          Username
        </h4>
        <input
          autoComplete="off"
          className="formInput"
          id="username"
          maxLength={16}
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="formLabel"
      >
        <h4>
          Password
        </h4>
        <input
          className="formInput"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      />
    </form>
  </div>
</div>
`;
